File: app/src/main/java/com/ebsoft/watchlist/di/AppComponent.java
package com.ebsoft.watchlist.di;

import android.app.Application;

import com.ebsoft.watchlist.WatchlistApplication;

import javax.inject.Singleton;

import dagger.BindsInstance;
import dagger.Component;
import dagger.android.support.AndroidSupportInjectionModule;

/**
 * Created by endre on 07/09/18.
 */

@Singleton
@Component(modules = {
        AndroidSupportInjectionModule.class,
        AppModule.class,
        FragmentBuilder.class,
        ActivityBuilder.class})
public interface AppComponent {

    @Component.Builder
    interface Builder {
        @BindsInstance
        Builder application(Application application);
        AppComponent build();
    }

    void inject(WatchlistApplication app);
}

File: app/src/main/java/com/ebsoft/watchlist/di/AppComponent.kt
package com.ebsoft.watchlist.di

import android.app.Application
import com.ebsoft.watchlist.WatchlistApplication
import dagger.BindsInstance
import dagger.Component
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton

/**
 * Created by endre on 07/09/18.
 */

@Singleton
@Component(modules = [
    AndroidSupportInjectionModule::class,
    AppModule::class,
    FragmentBuilder::class,
    ActivityBuilder::class])
interface AppComponent {

    @Component.Builder
    interface Builder {
        @BindsInstance
        fun application(application: Application): Builder
        fun build(): AppComponent
    }

    fun inject(app: WatchlistApplication)
}
